indexing-service/src/main/java/org/apache/druid/indexing/common/task/batch/parallel/ParallelIndexSupervisorTask.java

```diff
@@ -821,10 +836,10 @@ TaskStatus runHashPartitionMultiPhaseParallel(TaskToolbox toolbox) throws Except
       taskStatus = TaskStatus.failure(getId(), errMsg);
     }
 
-    toolbox.getTaskReportFileWriter().write(
-        getId(),
-        getTaskCompletionReports(taskStatus, segmentAvailabilityConfirmationCompleted)
-    );
+    completionReports = getTaskCompletionReports(taskStatus, segmentAvailabilityConfirmationCompleted);
+    if (!ingestionSchema.isSkipPublishingReports()) {
+      toolbox.getTaskReportFileWriter().write(getId(), completionReports);
```

Comment:
wondering if we actually need to do this, wouldn't the CompactionTask logic overwrite the file with the correct data?

Reply:
ah i guess that would be weird if there's an error